src/util/tryOrHandleMessageError.ts

```diff
@@ -0,0 +1,18 @@
+export function tryOrHandleMessageError(target: Function, context?: any): (x?: any) => any {
+  function tryCatcher() {
+    const { target, context } = (<any> tryCatcher);
+    try {
+      target.apply(context || this, arguments);
```

Comment:
@blesh would if I could, but `this` at line 5 won't be the same thing as `this` at line 16.